### ISSUE TITLE
Allows feature names that will result in length limits being exceed on deployment

### ISSUE THREAD

**Issue by otherjoel** [hougaard/Simple-Object-Designer#244]
I created a new Document-type feature named `Cx_Payment_Register` with the _Include Posted Variant_ option enabled, Deployment failed with this error:

```
Deployment failed. Errors: App ID : ab8e6875-5d1e-42cb-b326-da92f749c926 Message : { Extension compilation failed
Features/Cx_Payment_Register/Posted.Cx_Payment_Register.Header.Table.al(1,13): error AL0305: The length of the application object identifier 'Posted Cx_Payment_Register HeaderWin' cannot exceed 30 characters.

Features/Cx_Payment_Register/Posted.Cx_Payment_Register.Line.Table.al(1,13): error AL0305: The length of the application object identifier 'Posted Cx_Payment_Register LineWin' cannot exceed 30 characters.

Features/Cx_Payment_Register/Posted.Cx_Payment_Register.Subform.al(1,12): error AL0305: The length of the application object identifier 'Posted Cx_Payment_RegisterListPart' cannot exceed 30 characters.

Features/Cx_Payment_Register/Posted.Cx_Payment_Register.Line.APIPage.al(1,12): error AL0197: An application object of type 'Page' with name 'API Posted Cx_Payment_Register' is already declared by the extension 'Lotus by Winsted (1.0.1.29)'

Features/Cx_Payment_Register/Posted.Cx_Payment_Register.Header.APIPage.al(1,12): error AL0197: An application object of type 'Page' with name 'API Posted Cx_Payment_Register' is already declared by the extension 'Lotus by Winsted (1.0.1.29)' } - Job Id : 133b75b2-7488-4412-b0f5-7f2694f5be63
```

So while I was allowed to use Cx_Payment_Register as the name of the feature, the actual limit on the length of the name is governed by the longest identifier that SOD will generate from that name, which looks to be Posted Cx_Payment_Register HeaderWin. Since my “Object Prefex” setting is 3 characters long and `Posted ` and ` Header` are 14 characters long combined (including leading/trailing spaces) it seems my feature name must not exceed 30-(14+3) = 13 characters.

Maybe this detailed length limit check can be included in the "Verify" function in a future release.

**Comment by otherjoel** [comment]
I normally would try to verify that shortening the name of my feature would fix the problem, but I can’t until #245 is resolved.

**Comment by otherjoel** [comment]
Shortening the name to under 13 characters solved my problem; I’m leaving this open in case you want to track it, but you can close it now if you wish.
